lib/Parse/ParseExpr.cpp

```diff
@@ -3185,7 +3185,7 @@ ParserResult<Expr> Parser::parseExprCollection() {
   if (Tok.is(tok::r_square)) {
     if (SyntaxContext->isEnabled())
       SyntaxContext->addSyntax(
-          SyntaxFactory::makeBlankArrayElementList(&Context.getSyntaxArena()));
+          SyntaxFactory::makeBlankArrayElementList(Context.getSyntaxArena()));
```

Comment:
Now that `SyntaxContext` owns `SyntaxArena`, I think we can rewrite all  `Context.getSyntaxArena()` in Parser with `SyntaxContext->getArena()`. (in followups is OK).

Reply:
I'm going to make a follow-up PR for this sometime next week.